File: languages/java/src/main/java/com/bloock/sdk/entity/EnsSigner.java
package com.bloock.sdk.entity;

import com.bloock.sdk.bridge.proto.RecordOuterClass;
import com.bloock.sdk.bridge.proto.RecordOuterClass.SignerAlg;

public class EnsSigner implements Signer {
  RecordOuterClass.SignerAlg alg;
  SignerArgs args;

  public EnsSigner(String privateKey) {
    this.alg = SignerAlg.ENS;
    this.args = new SignerArgs(privateKey);
  }

  @Override
  public RecordOuterClass.Signer toProto() {
    return RecordOuterClass.Signer.newBuilder()
        .setAlg(this.alg)
        .setArgs(this.args.toProto())
        .build();
  }
}

File: languages/php/Bloock/Entity/EnsSigner.php
<?php

namespace Bloock\Entity;

use Bloock\SignerAlg;

class EnsSigner implements Signer
{
    public int $alg;
    public SignerArgs $args;

    public function __construct(string $privateKey) {
        $this->alg = SignerAlg::ENS;
        $this->args = new SignerArgs($privateKey);
    }

    public function toProto(): \Bloock\Signer {
        $s = new \Bloock\Signer();
        $s->setAlg($this->alg);
        $s->setArgs($this->args->toProto());
        return $s;
    }
}